plugin/helpers_bots.go

```diff
@@ -9,6 +9,39 @@ import (
 	"github.com/pkg/errors"
 )
 
+type shouldProcessMessageOptions struct {
+	AllowSystemMessages bool
+	AllowBots           bool
+	FilterChannelIDs    []string
+	FilterUserIDs       []string
+}
+
+type ShouldProcessMessageOption func(*shouldProcessMessageOptions)
+
+func AllowSystemMessages() ShouldProcessMessageOption {
+	return func(options *shouldProcessMessageOptions) {
+		options.AllowSystemMessages = true
+	}
+}
+
+func FilterChannelIDs(filterChannelIDs []string) ShouldProcessMessageOption {
```

Comment:
nit: it might be nice to order these methods to match their declaration in the `shouldProcessMessageOptions`. Purely for readability.